src/xcode/ENA/ENA/Source/Workers/Logging.swift

```diff
@@ -0,0 +1,149 @@
+import Foundation
+import os.log
+
+extension OSLog {
+
+	private static var subsystem = Bundle.main.unwrappedBundleIdentifier
+
+    /// API interactions
+    static let api = OSLog(subsystem: subsystem, category: "api")
+    /// Exoplanet UI
+    static let ui = OSLog(subsystem: subsystem, category: "ui")
+    /// Local data & caches
+    static let localData = OSLog(subsystem: subsystem, category: "localdata")
+}
+
+enum Log {
+
+    static func debug(_ message: String, log: OSLog = .default) {
+        Self.log(message: message, type: .debug, log: log, error: nil)
+    }
+
+    static func info(_ message: String, log: OSLog = .default) {
+        Self.log(message: message, type: .info, log: log, error: nil)
+    }
+
+    static func warning(_ message: String, log: OSLog = .default) {
+        Self.log(message: message, type: .default, log: log, error: nil)
+    }
+
+    static func error(_ message: String, log: OSLog = .default, error: Error? = nil) {
+        Self.log(message: message, type: .error, log: log, error: error)
+    }
+
+	private static func log(message: String, type: OSLogType, log: OSLog, error: Error?) {
+		os_log("%{private}@", log: log, type: type, message)
```

Comment:
AFAIK we are not allowed to even to os_log in Production

Reply:
But everything is "private", i.e. log output of the whole `message` will be:
`DATE_FOO <private>`

Reply:
What I am wondering currently, how do we differentiate between TestFlight and AppStore builds?

Reply:
`RELEASE` seems to be set for release builds. So:

```
#if !RELEASE
// log log log!
#endif
```

Reply:
Wouldn't this be the same for TestFlight Builds?

Reply:
TestFlight builds are build with "TestFlight" configuration. And this configuration does not have a RELEASE flag configured in the build settings. So my guess is, logs would work for TestFlight releases.

<img width="713" alt="Bildschirmfoto 2020-10-09 um 10 01 25" src="https://user-images.githubusercontent.com/1941000/95558527-a17fd600-0a16-11eb-993a-cb3466b98121.png">

Reply:
I have "hidden" os_log call behind !RELEASE